### DISCUSSION TITLE
Sales: and/or condition for categories

### DISCUSSION THREAD

**Discussion by billmn** [craftcms/commerce#2464] [Ideas]
**What are you trying to do?**
I have, for instance, two categories: "Brands" and "Gender".
I would apply a sale on "Nike / Men" (e.g. 10% discount) and a different sale on "Nike / Women" (e.g. 20% discount).

Now is not possible because it uses an OR condition between categories.

**What's your proposed solution?**
What about add a field in "Sale -> Matching Items" section that allow to choose the and/or condition?

"OR" condition: limit the discount to one or more categories (as default, like now)
"AND" condition: limit the discount to the combination of all specified categories

Could be enough change the `relatedTo` call here?
https://github.com/craftcms/commerce/blob/692211a67c7aac90f2af2bdb1d2cbb7172684222/src/services/Sales.php#L500

**Comment by lukeholder** [comment]
Thanks for the request. We do have plans to add features like this to sales, but it is not as simple as just modifying that query as we also support the `hasSales` variant query param which is more complex.

**Comment by billmn** [comment]
Thank you @lukeholder for the answer!
Hope this feature arrives soon 🤞 😉

**Comment by mcclaskiem** [comment]
+1 this, this is a pretty critical feature for a few of my Commerce clients. Almost every ecommerce site has multiple category groups for their products and to be able to fine tune sales to multiple categories would be extremely helpful.
